library/ssl_srv.c

```diff
@@ -3585,15 +3585,19 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     /* Read the message without adding it to the checksum */
     do {
 
-        if( ( ret = mbedtls_ssl_read_record_layer( ssl ) ) != 0 )
+        do ret = mbedtls_ssl_read_record_layer( ssl );
+        while( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING );
```

Comment:
@hanno-arm I missed that `MBEDTLS_ERR_SSL_CONTINUE_PROCESSING ` is returned by `mbedtls_ssl_read_record_layer ` only in case of DTLS, so instead of `goto read_record_header`, which is ugly looping, you do the loop at a one higher level. I retract my comment